app/uk/gov/hmrc/gform/gform/DownloadController.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.gform
+
+import play.api.Environment
+import play.api.mvc.{ Action, AnyContent, MessagesControllerComponents }
+import uk.gov.hmrc.gform.config.FileInfoConfig
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.ExecutionContext
+
+class DownloadController(messagesControllerComponents: MessagesControllerComponents, environment: Environment)(implicit
+  ec: ExecutionContext
+) extends FrontendController(messagesControllerComponents) {
+
+  def downloadNipTemplate(fileExt: String): Action[AnyContent] = Action { _ =>
+    val fileName = s"nipClaimScheduleTemplate.$fileExt"
```

Comment:
I think it is potentially dangerous. `fileExt` is a user input. They can provide a path instead of extension and potentially traverse our file system and get access to other files.
Can we sanitize it, like checking that it is in the list of valid extensions?